src/f0/app_solar/source/tmp101.c

```diff
@@ -0,0 +1,261 @@
+/* Driver local functions.                                                   */
+/*===========================================================================*/
+
+#if (TMP101_USE_I2C) || defined(__DOXYGEN__)
+/**
+ * @brief   Reads registers value using I2C.
+ * @pre     The I2C interface must be initialized and the driver started.
+ *
+ * @param[in]  i2cp      pointer to the I2C interface
+ * @param[in]  sad       slave address without R bit
+ * @param[in]  reg       first sub-register address
+ * @param[out] rxbuf     pointer to an output buffer
+ * @param[in]  n         number of consecutive register to read
+ * @return               the operation status.
+ * @notapi
+ */
+msg_t tmp101I2CReadRegister(I2CDriver *i2cp, i2caddr_t sad, uint8_t reg,
+        uint8_t* rxbuf, size_t n) {
+    return i2cMasterTransmitTimeout(i2cp, sad, &reg, 1, rxbuf, n,
+            TIME_INFINITE);
+}
+
+/**
+ * @brief   Writes a value into a register using I2C.
+ * @pre     The I2C interface must be initialized and the driver started.
+ *
+ * @param[in] i2cp       pointer to the I2C interface
+ * @param[in] sad        slave address without R bit
+ * @param[in] txbuf      buffer containing reg in first byte and high
+ *                       and low data bytes
+ * @param[in] n          size of txbuf
+ * @return               the operation status.
+ * @notapi
+ */
+msg_t tmp101I2CWriteRegister(I2CDriver *i2cp, i2caddr_t sad, uint8_t *txbuf,
+        size_t n) {
+    return i2cMasterTransmitTimeout(i2cp, sad, txbuf, n, NULL, 0,
+            TIME_INFINITE);
+}
+#endif /* TMP101_USE_I2C */
+
+/*==========================================================================*/
+/* Interface implementation.                                                */
+/*==========================================================================*/
+
+static const struct TMP101VMT vmt_device = {
+    (size_t)0,
+};
+
+/*===========================================================================*/
+/* - SECTION - Driver exported functions.                                    */
+/*===========================================================================*/
+
+/**
+ * @brief   Initializes an instance.
+ *
+ * @param[out] devp     pointer to the @p TMP101Driver object
+ *
+ * @init
+ */
+void tmp101ObjectInit(TMP101Driver *devp) {
+    devp->vmt = &vmt_device;
+
+    devp->config = NULL;
+
+    devp->state = TMP101_STOP;
+}
+
+
+
+
+/**
+ * @brief   Configures and activates TMP101 Complex Driver peripheral.
+ *
+ * @param[in] devp      pointer to the @p TMP101Driver object
+ * @param[in] config    pointer to the @p TMP101Config object
+ *
+ * @api
+ */
+void tmp101Start(TMP101Driver *devp, const TMP101Config *config) {
+//    i2cbuf_t buf;
+
+    osalDbgCheck((devp != NULL) && (config != NULL));
+    osalDbgAssert((devp->state == TMP101_STOP) ||
+            (devp->state == TMP101_READY),
+            "tmp101Start(), invalid state");
+
+    devp->config = config;
+
```

Comment:
It does not look like this code properly starts the I2C drivers (i2cStart). Take a look at the INA226 code for the proper starting and stopping of the I2C driver from the respective Start and Stop functions in this driver.

Reply:
Code restored here to model the pattern in INA226 driver.  That code now more clear to pull requester.